File: sportseq-admin/src/main/java/com/company/sportseq/entity/Equipment.java
package com.company.sportseq.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.company.sportseq.common.domain.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("equipment")
public class Equipment extends BaseEntity {

    @TableId(type = IdType.AUTO)
    private Long id;
    private String equipmentCode;
    private String equipmentName;
    private Long categoryId;
    private String brand;
    private String model;
    private String spec;
    private String unit;
    private BigDecimal purchasePrice;
    private Integer safeStock;
    private Integer maxBorrowDays;
    private String imageUrl;
    private Integer status;
    private String description;
}

File: sportseq-web/src/api/equipment/equipment.ts
import { del, get, post, put } from '../request'
import type { PageResult } from '../request'

export interface EquipmentItem {
  id: number
  equipmentCode: string
  equipmentName: string
  categoryId: number
  categoryName: string | null
  brand: string | null
  model: string | null
  spec: string | null
  unit: string
  purchasePrice: number | null
  safeStock: number
  maxBorrowDays: number | null
  imageUrl: string | null
  status: number
  description: string | null
  createTime: string
}

export interface EquipmentDTO {
  id?: number
  equipmentName: string
  categoryId: number
  brand?: string
  model?: string
  spec?: string
  unit: string
  purchasePrice?: number
  safeStock?: number
  maxBorrowDays?: number
  imageUrl?: string
  status?: number
  description?: string
}

export interface ImportResult {
  successCount: number
  failCount: number
  errors: string[]
}

export const getEquipmentPage = (params: {
  current: number
  size: number
  equipmentName?: string
  equipmentCode?: string
  categoryId?: number
  status?: number
}) => get<PageResult<EquipmentItem>>('/equipment/page', { params })

export const getEquipmentDetail = (id: number) => get<EquipmentItem>(`/equipment/${id}`)

export const addEquipment = (data: EquipmentDTO) => post<void>('/equipment', data)

export const updateEquipment = (data: EquipmentDTO) => put<void>('/equipment', data)

export const deleteEquipment = (id: number) => del<void>(`/equipment/${id}`)

export const changeEquipmentStatus = (id: number, status: number) =>
  put<void>(`/equipment/${id}/status?status=${status}`)

export const importEquipment = (file: File) => {
  const form = new FormData()
  form.append('file', file)
  return post<ImportResult>('/equipment/import', form, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}

export const exportEquipmentUrl = (params: {
  equipmentName?: string
  equipmentCode?: string
  categoryId?: number
  status?: number
}) => `/equipment/export?${new URLSearchParams(
  Object.entries(params).filter(([, v]) => v !== undefined && v !== '') as [string, string][],
).toString()}`
